pkg/nrtupdater/node_getter.go

```diff
@@ -0,0 +1,68 @@
+package nrtupdater
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NotConfigured struct{}
+
+func (err NotConfigured) Error() string {
+	return "unconfigured feature"
+}
+
+type ConnectionError struct {
+	Err error
+}
+
+func (err ConnectionError) Error() string {
+	return fmt.Sprintf("error connection k8s: %v", err.Err)
+}
+func (err ConnectionError) Unwrap() error {
+	return err.Err
+}
+
+func IsNotConfiguredNodeGetter(err error) bool {
+	_, ok := err.(NotConfigured)
+	return ok
+}
```

Comment:
we probably want to use the more modern idiom `errors.Is` https://go.dev/blog/go1.13-errors

Reply:
Thanks for the link